fsspec/implementations/http.py

```diff
@@ -847,6 +847,11 @@ async def _file_info(url, session, size_policy="head", **kwargs):
         elif "Content-Range" in r.headers:
             info["size"] = int(r.headers["Content-Range"].split("/")[1])
 
+        if "Content-Type" in r.headers:
+            info["mime"] = r.headers["Content-Type"].partition(";")[0]
```

Comment:
It is possible this header doesn't exist

Reply:
This is presumably taken care of by the `if` statement.
Would you prefer `mime` was set to `None` if the header was missing?

Reply:
Ah indeed. First day back...
By the way, what does the ";" do, how do you expect this to format?

Reply:
The [`Content-Type`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Type) HTTP header can take the following form:

```
Content-Type: text/html; charset=utf-8
Content-Type: multipart/form-data; boundary=something
```

so partitioning on the first `;` selects the mimetype from the value

Reply:
OK. Is there any need for verification on that? I suppose we can just pass on the string and any client might do that.

Reply:
I agree it makes sense to just pass on the string and leave verification to the client